### PR TITLE
Datastore support read only

### PR DESCRIPTION
**Pull request by fjnoyp**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/amplify_api/lib/src/method_channel_api.dart [hunk 1]
```diff
@@ -251,6 +251,9 @@ class AmplifyAPIMethodChannel extends AmplifyAPI {
     inputsMap['restOptions'] = restOptions.serializeAsMap();
     inputsMap['cancelToken'] = cancelToken;
 
+    // Refresh auth tokens
+    await _authProviderRefresher();
```
Comment (HuiSF) [inline packages/amplify_api/lib/src/method_channel_api.dart:255]: This change has been reverted on main, which should not be brought into RC. Please remove corresponding commit.

**Review by HuiSF — CHANGES_REQUESTED** [review]

File: packages/amplify_api/lib/src/method_channel_api.dart (hunk 1 above)
Comment (fjnoyp) [reply packages/amplify_api/lib/src/method_channel_api.dart]: Ah thank you.  Reverted the corresponding commit.

**Review by HuiSF — APPROVED** [review]

File: packages/amplify_datastore_plugin_interface/test/testData/Person.dart [hunk 2]
```diff
@@ -13,30 +13,21 @@
...
+// ignore_for_file: public_member_api_docs
+
+import 'ModelProvider.dart';
+
 /** This is an auto generated class representing the Person type in your schema. */
 @immutable
-class Person extends Model {
-  static const classType = const _PersonModelType();
-  final String id;
+class Person {
```
Comment (dnys1) [inline packages/amplify_datastore_plugin_interface/test/testData/Person.dart:26]: What's up with this class

**Review by dnys1 — APPROVED** [review]

File: packages/amplify_datastore_plugin_interface/test/testData/Person.dart (hunk 2 above)
Comment (HuiSF) [reply packages/amplify_datastore_plugin_interface/test/testData/Person.dart:26]: Oops missed this one, seems the model generated by 

```
type Person {
  name: String!
  propertiesAddresses: [Address]
  contact: Contact!
}
```

which should be with `@model`.

**Pull request merged**